File: app/src/test/java/com/github/gfx/helium/HatebuFeedTest.java
package com.github.gfx.helium;

import com.github.gfx.helium.model.HatebuEntry;
import com.github.gfx.helium.model.HatebuFeed;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.simpleframework.xml.Serializer;
import org.simpleframework.xml.core.Persister;

import android.support.test.runner.AndroidJUnit4;

import java.io.InputStream;

import static com.github.gfx.helium.TestUtils.openAsset;
import static org.hamcrest.MatcherAssert.*;
import static org.hamcrest.Matchers.*;

@RunWith(AndroidJUnit4.class)
public class HatebuFeedTest {

    @Test
    public void parseHatebuHotentryFeed() throws Exception {
        Serializer serializer = new Persister();
        InputStream stream = openAsset("hotentries.rss");
        HatebuFeed feed = serializer.read(HatebuFeed.class, stream);
        stream.close();

        assertThat(feed.items.size(), is(30));

        HatebuEntry entry = feed.items.get(0);

        assertThat(entry.title, is(not("")));
        assertThat(entry.title, is(notNullValue()));

        assertThat(entry.description, is(not("")));
        assertThat(entry.description, is(notNullValue()));

        assertThat(entry.subject, hasSize(1));

        assertThat(entry.getTimestamp(), is(notNullValue()));
    }

    @Test
    public void parseHatebuFavoriteFeed() throws Exception {
        Serializer serializer = new Persister();
        InputStream stream = openAsset("favorites.rss");
        HatebuFeed feed = serializer.read(HatebuFeed.class, stream);
        stream.close();

        assertThat(feed.items.size(), is(25));

        HatebuEntry entry = feed.items.get(0);

        assertThat(entry.title, is(not("")));
        assertThat(entry.title, is(notNullValue()));

        assertThat(entry.description, is(not("")));
        assertThat(entry.description, is(notNullValue()));

        assertThat(entry.subject, hasSize(0));

        assertThat(entry.getTimestamp(), is(notNullValue()));

        assertThat(entry.creator, is(notNullValue()));
    }
}

File: app/src/test/java/com/github/gfx/helium/HatebuFeedTest.kt
package com.github.gfx.helium

import com.github.gfx.helium.model.HatebuEntry
import com.github.gfx.helium.model.HatebuFeed

import org.junit.Test
import org.junit.runner.RunWith
import org.simpleframework.xml.Serializer
import org.simpleframework.xml.core.Persister

import android.support.test.runner.AndroidJUnit4

import java.io.InputStream

import com.github.gfx.helium.TestUtils.openAsset
import org.hamcrest.MatcherAssert.*
import org.hamcrest.Matchers.*

@RunWith(AndroidJUnit4::class)
class HatebuFeedTest {

    @Test
    @Throws(Exception::class)
    fun parseHatebuHotentryFeed() {
        val serializer = Persister()
        val stream = openAsset("hotentries.rss")
        val feed = serializer.read<HatebuFeed>(HatebuFeed::class.java, stream)
        stream.close()

        assertThat(feed.items.size, `is`(30))

        val entry = feed.items[0]

        assertThat(entry.title, `is`(not("")))
        assertThat(entry.title, `is`(notNullValue()))

        assertThat(entry.description, `is`(not("")))
        assertThat(entry.description, `is`(notNullValue()))

        assertThat(entry.subject, hasSize<Any>(1))

        assertThat(entry.timestamp, `is`(notNullValue()))
    }

    @Test
    @Throws(Exception::class)
    fun parseHatebuFavoriteFeed() {
        val serializer = Persister()
        val stream = openAsset("favorites.rss")
        val feed = serializer.read<HatebuFeed>(HatebuFeed::class.java, stream)
        stream.close()

        assertThat(feed.items.size, `is`(25))

        val entry = feed.items[0]

        assertThat(entry.title, `is`(not("")))
        assertThat(entry.title, `is`(notNullValue()))

        assertThat(entry.description, `is`(not("")))
        assertThat(entry.description, `is`(notNullValue()))

        assertThat(entry.subject, hasSize<Any>(0))

        assertThat(entry.timestamp, `is`(notNullValue()))

        assertThat(entry.creator, `is`(notNullValue()))
    }
}
